docs/template_plugin/tests/functional/op_reference/gather_tree.cpp

```diff
@@ -3,91 +3,144 @@
 #include "base_reference_test.hpp"
+#include "ngraph/runtime/reference/gather_tree.hpp"
+#include "ngraph_functions/utils/data_utils.hpp"
 
 using namespace reference_tests;
 using namespace ov;
 
 namespace {
 struct GatherTreeParams {
-    template <class IN_ET>
-    GatherTreeParams(const ov::Shape inShape, std::vector<IN_ET> stepIds, const std::vector<IN_ET> parentIds,
-        const std::vector<IN_ET> maxSeqLen, const std::vector<IN_ET> endToken, std::vector<IN_ET> output) :
-        stepIdsTensor(inShape, element::from<IN_ET>(), stepIds), parentIdsTensor(inShape, element::from<IN_ET>(), parentIds),
-        maxSeqLenTensor(ov::Shape{inShape[1]}, element::from<IN_ET>(), maxSeqLen), endTokenTensor(ov::Shape{}, element::from<IN_ET>(), endToken),
-        expectedTensor(inShape, element::from<IN_ET>(), output) {}
-    Tensor stepIdsTensor;
-    Tensor parentIdsTensor;
-    Tensor maxSeqLenTensor;
-    Tensor endTokenTensor;
-    Tensor expectedTensor;
+    Tensor stepIds;
+    Tensor parentIdx;
+    Tensor maxSeqLen;
+    Tensor endToken;
+    Tensor finalIdx;
+    std::string testcaseName;
+};
+
+struct Builder : ParamsBuilder<GatherTreeParams> {
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, stepIds);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, parentIdx);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, maxSeqLen);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, endToken);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, finalIdx);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, testcaseName);
 };
 
 class ReferenceGatherTreeTest : public testing::TestWithParam<GatherTreeParams>, public CommonReferenceTest {
 public:
     void SetUp() override {
         auto params = GetParam();
         function = CreateFunction(params);
-        inputData = {params.stepIdsTensor.data, params.parentIdsTensor.data, params.maxSeqLenTensor.data, params.endTokenTensor.data};
-        refOutData = {params.expectedTensor.data};
+        inputData = {params.stepIds.data, params.parentIdx.data, params.maxSeqLen.data, params.endToken.data};
+        refOutData = {params.finalIdx.data};
     }
+
     static std::string getTestCaseName(const testing::TestParamInfo<GatherTreeParams>& obj) {
         auto param = obj.param;
         std::ostringstream result;
-        result << "iType=" << param.stepIdsTensor.type << "_";
-        result << "iShape=" << param.stepIdsTensor.shape;
+        result << "sType=" << param.stepIds.type;
+        result << "_sShape=" << param.stepIds.shape;
+        result << "_pType=" << param.parentIdx.type;
+        result << "_pShape=" << param.parentIdx.shape;
+        result << "_mType=" << param.maxSeqLen.type;
+        result << "_mShape=" << param.maxSeqLen.shape;
+        result << "_eType=" << param.endToken.type;
+        result << "_eShape=" << param.endToken.shape;
+        result << "_fType=" << param.finalIdx.type;
+        if (!param.testcaseName.empty()) {
+            result << "_fShape=" << param.finalIdx.shape;
```

Comment:
I think this line can be out of if statement, right? Sorry, I find it now.

Reply:
No problem, I missed as well. And it's updated.